nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ListenSMTP.java

```diff
@@ -13,89 +13,52 @@
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
 
+import org.apache.commons.io.IOUtils;
 import org.apache.nifi.annotation.behavior.InputRequirement;
-import org.apache.nifi.annotation.behavior.WritesAttribute;
-import org.apache.nifi.annotation.behavior.WritesAttributes;
 import org.apache.nifi.annotation.documentation.CapabilityDescription;
 import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnStopped;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.AbstractSessionFactoryProcessor;
+import org.apache.nifi.processor.DataUnit;
 import org.apache.nifi.processor.ProcessContext;
 import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.ProcessSessionFactory;
+import org.apache.nifi.processor.Relationship;
 import org.apache.nifi.processor.exception.ProcessException;
-import org.apache.nifi.components.ValidationContext;
-import org.apache.nifi.components.ValidationResult;
-import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.io.OutputStreamCallback;
+import org.apache.nifi.processor.util.StandardValidators;
 import org.apache.nifi.ssl.SSLContextService;
-
-import org.apache.nifi.processors.email.smtp.event.SmtpEvent;
-import org.apache.nifi.processors.email.smtp.handler.SMTPResultCode;
-import org.apache.nifi.processors.email.smtp.handler.SMTPMessageHandlerFactory;
+import org.subethamail.smtp.server.SMTPServer;
 
 @Tags({"listen", "email", "smtp"})
 @InputRequirement(InputRequirement.Requirement.INPUT_FORBIDDEN)
-@CapabilityDescription("This processor implements a lightweight SMTP server to an arbitrary port, " +
-        "allowing nifi to listen for incoming email. " +
-        "" +
-        "Note this server does not perform any email validation. If direct exposure to the internet is sought," +
-        "it may be a better idea to use the combination of NiFi and an industrial scale MTA (e.g. Postfix)")
-@WritesAttributes({
-        @WritesAttribute(attribute = "mime.type", description = "The value used during HELO"),
-        @WritesAttribute(attribute = "smtp.helo", description = "The value used during HELO"),
-        @WritesAttribute(attribute = "smtp.certificates.*.serial", description = "The serial numbers for each of the " +
-                "certificates used by an TLS peer"),
-        @WritesAttribute(attribute = "smtp.certificates.*.principal", description = "The principal for each of the " +
-                "certificates used by an TLS peer"),
-        @WritesAttribute(attribute = "smtp.from", description = "The value used during MAIL FROM (i.e. envelope)"),
-        @WritesAttribute(attribute = "smtp.to", description = "The value used during RCPT TO (i.e. envelope)"),
-        @WritesAttribute(attribute = "smtp.src", description = "The source IP of the SMTP connection")})
```

Comment:
As I just mentioned in the previous moment, adding things in the future is possible, removing is not. So I'd like to know (outside of "just in case") practical implication of those attributes to justify having them now. Also, the to/from/and few others can be derived from message.